ckanext/qa/sniff_format.py

```diff
@@ -109,15 +122,14 @@ def sniff_file_format(filepath):
 
         if format_['format'] == 'TXT':
             # is it JSON?
-            with open(filepath, 'rU') as f:
-                buf = f.read(10000)
+            buf = read_unknown_encoding(filepath, 10000, mode='rtU')
             if is_json(buf):
                 format_ = {'format': 'JSON'}
             # is it CSV?
```

Comment:
Eh, PSV and CSV are variations on the same thing, really, just a different delimiter (PSV is pipe-separated values). They even use the same library to attempt parsing them, and the reason I had to swap them is because that library now recognises pipes by default as a valid form of CSV.